stablehlo/reference/Scope.cpp

```diff
@@ -0,0 +1,60 @@
+/* Copyright 2023 The StableHLO Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "Scope.h"
+
+#include "llvm/Support/FormatVariadic.h"
+#include "mlir/Support/DebugStringHelper.h"
+
+namespace mlir {
+namespace stablehlo {
+
+void Scope::add(Value ssaValue, Tensor runtimeValue) {
+  // We are instantiating a new `Scope` object every time the
+  // interpreter evaluates a region. With that, the `stackFrame` should not have
+  // any duplicates.
+  assert(!stackFrame.count(ssaValue));  // TODO
```

Comment:
What does this TODO mean?

Reply:
The Todo is related to  adding a  report_fatal_error error. Done.